model/src/cmd/ModelCommandAPI.cc

```diff
@@ -0,0 +1,369 @@
+                          + std::to_string(inertialMatrix.Iyz()) + "]\n"
+              "        [" + std::to_string(inertialMatrix.Ixz()) + " | "
+                          + std::to_string(inertialMatrix.Iyz()) + " | "
+                          + std::to_string(inertialMatrix.Izz()) + "]";
+      std::cout << "    - Mass: " << massInfo << std::endl
+                << "    - Inertial Matrix: " << inertialInfo << std::endl;
+    }
+
+    const auto poseComp =
+        _ecm.Component<ignition::gazebo::components::Pose>(entity);
+
+    if (poseComp)
+    {
+      const std::string poseInfo =
+            "\n        [" + std::to_string(poseComp->Data().X()) + " | "
+                          + std::to_string(poseComp->Data().Y()) + " | "
+                          + std::to_string(poseComp->Data().Z()) + "]\n"
+              "        [" + std::to_string(poseComp->Data().Roll()) + " | "
+                          + std::to_string(poseComp->Data().Pitch()) + " | "
+                          + std::to_string(poseComp->Data().Yaw()) + "]";
+
+      std::cout << "    - Pose: " << poseInfo << std::endl;
+    }
+  }
+}
+
+//////////////////////////////////////////////////
+// \brief Print the model joints information.
+// \param[in] _entity Entity of the model requested.
+// \param[in] _ecm ECM ready for requests.
+// \param[in] _jointName Joint to be printed, if nullptr, print all joints.
+void printJoints(const uint64_t entity,
+                const ignition::gazebo::EntityComponentManager &_ecm,
+                const std::string &_jointName)
+{
+  static const std::map<sdf::JointType, std::string> jointTypes = {
+    {sdf::JointType::REVOLUTE, "revolute"},
+    {sdf::JointType::BALL, "ball"},
+    {sdf::JointType::CONTINUOUS, "continuous"},
+    {sdf::JointType::FIXED, "fixed"},
+    {sdf::JointType::GEARBOX, "gearbox"},
+    {sdf::JointType::PRISMATIC,  "prismatic"},
+    {sdf::JointType::REVOLUTE2, "revolute2"},
+    {sdf::JointType::SCREW, "screw"},
+    {sdf::JointType::UNIVERSAL, "universal"}
+  };
+
+  const auto joints = _ecm.EntitiesByComponents(
+  ignition::gazebo::components::ParentEntity(entity),
+  ignition::gazebo::components::Joint());
+
+  for(const auto &_entity : joints)
+  {
+    const auto parentComp =
+        _ecm.Component<ignition::gazebo::components::ParentEntity>(_entity);
+
+    const auto nameComp =
+        _ecm.Component<ignition::gazebo::components::Name>(_entity);
+
+    if(_jointName.length() && _jointName != nameComp->Data())
+        continue;
+
+    if (parentComp)
+    {
+      std::string parentInfo;
+      auto parentNameComp =
+          _ecm.Component<ignition::gazebo::components::Name>(
+          parentComp->Data());
+
+      if (parentNameComp)
+      {
+        parentInfo += parentNameComp->Data() + " ";
+      }
+      parentInfo += "[" + std::to_string(parentComp->Data()) + "]";
+
+      std::cout << "  - Joint [" << _entity << "]" << std::endl
```

Comment:
I see that the indentation for all messages is hardcoded right now. I think this is fine for `ign-gazebo3`, because we don't have good nested model support. But from `ign-gazebo5`, printing information about nested models can become a bit messy with a fixed indentation.

One idea to tackle this would be to accept the indentation as a parameter in the `print*` functions, and then you print the necessary whitespace before each line as needed.

This doesn't need to be tackled here, I think we could open an issue for it and tackle it in the future.

Reply:
The implementation done in this PR isn't taking into account the nested model possibility, so a new Issue will be created once this one is merged